crates/nu-protocol/src/shell_error.rs

```diff
@@ -801,8 +801,11 @@ pub enum ShellError {
     ///
     /// Make sure the directory in the error message actually exists before trying again.
     #[error("Directory not found")]
-    #[diagnostic(code(nu::shell::directory_not_found))]
-    DirectoryNotFound(#[label("directory not found")] Span, #[help] Option<String>),
+    #[diagnostic(
+        code(nu::shell::directory_not_found),
+        help("the target directory was {0}")
+    )]
+    DirectoryNotFound(String, #[label("directory not found")] Span, Option<String>),
```

Comment:
What about change the definition like this:
```rust
   #[diagnostic(
        code(nu::shell::directory_not_found),
        help("{1} not found")
    )]
    DirectoryNotFound(String, #[label("directory not found")] Span, String),
```

We already defined it's `DirectoryNotFound`, I don't think we need to provide underlying `IO::Error` message to user.
So the error message will be something like this:
```nushell
 × Directory not found
   ╭─[entry #1:1:1]
 1 │ ["a.txt"] | each {|$it| open $it}
   ·                              ─┬─
   ·                               ╰── directory not found
   ╰────
  help: /Users/tmp/projects/rust_online_code/nushell/a.txt not found
```
Which is less verbosing.

Reply:
should be better in b73b30424 :yum: 

just need to fix the tests now i guess :wink: